packages/analytics-js/src/plugins/eventProcessor/eventProcessor.ts

```diff
@@ -0,0 +1,74 @@
+import { ExtensionPlugin } from '@rudderstack/analytics-js/npmPackages/js-plugin/types';
+import {
+  RudderEvent,
+  RudderEventType,
+} from '@rudderstack/analytics-js/components/eventManager/types';
+import { state } from '@rudderstack/analytics-js/state';
+import { ApiObject, ApiOptions } from '@rudderstack/analytics-js/state/types';
+import { getCurrentTimeFormatted } from '@rudderstack/analytics-js/components/utilities/timestamp';
+import { generateUUID } from '@rudderstack/analytics-js/components/utilities/uuId';
+import { mergeDeepRight } from '@rudderstack/analytics-js/components//utilities/object';
+import { defaultLogger } from '@rudderstack/analytics-js/services/Logger';
+import { Nullable } from '@rudderstack/analytics-js/types';
+import { CHANNEL } from './constants';
+import { checkForReservedElements, getContextPageProperties, processOptions } from './utilities';
+
+const eventProcessor = (): ExtensionPlugin => ({
+  name: 'eventProcessor',
+  deps: [],
+  event: {
+    /**
+     * Appends the common event data to the rudder event
+     * @param rudderEvent Generate rudder event
+     * @param pageProps Page properties
+     * @returns Complete Rudder Event object
+     */
+    process: (
+      rudderEvent: Partial<RudderEvent>,
+      options?: Nullable<ApiOptions>,
+      pageProps?: ApiObject,
+    ): RudderEvent => {
+      const commonEventData = {
+        // Type casting to string as the user session manager will take care of initializing the value
+        anonymousId: state.session.rl_anonymous_id.value as string,
+        channel: CHANNEL,
+        context: {
+          traits: { ...state.session.rl_trait.value },
```

Comment:
should we clone() instead of shallow copy to avoid issues with nested objects if we allow custom event payload mutations in the future?

Reply:
Addressed.